### PR TITLE
Creator dashboard project activity loading indicator

### PR DESCRIPTION
**Pull request by cdolm92**
# What

Added a loading indicator when navigating to dashboard tab.

# Why

Before adding the loading indicator the user would encounter a blank screen as the dashboard was loading.  The loading indicator gives the user the feedback that the dashboard is loading.

# See 👀

loader

### CODE REVIEW

File: Kickstarter-iOS/Views/Controllers/DashboardViewController.swift [hunk 1]
```diff
@@ -23,6 +25,10 @@ internal final class DashboardViewController: UITableViewController {
   internal override func viewDidLoad() {
     super.viewDidLoad()
 
+    self.tableView.backgroundView = self.backgroundView
```
Comment (justinswart) [inline Kickstarter-iOS/Views/Controllers/DashboardViewController.swift]: This approach feels like we're fighting `UIKit`. I know `UITableViewController`s can be pretty stubborn when it comes to subviews, but I'm wondering if it wouldn't be better to do this using an empty state or a cell type that we can display while loading.

File: Kickstarter-iOS/Views/Controllers/DashboardViewController.swift [hunk 2]
```diff
@@ -52,9 +65,16 @@ internal final class DashboardViewController: UITableViewController {
     self.viewModel.inputs.viewWillDisappear()
   }
 
+  internal override func viewDidLayoutSubviews() {
+    super.viewDidLayoutSubviews()
+
+    self.loadingIndicatorView.center = self.backgroundView.center
```
Comment (justinswart) [inline Kickstarter-iOS/Views/Controllers/DashboardViewController.swift]: I think we should center using layout constraints where possible.

File: Library/ViewModels/DashboardViewModel.swift [hunk 3]
```diff
@@ -148,6 +154,11 @@ public final class DashboardViewModel: DashboardViewModelInputs, DashboardViewMo
 
     self.project = projectsAndSelected.map(second)
 
+    self.loaderIsAnimating = Signal.merge(
+    //  self.viewDidLoadProperty.signal.map(const(true)),
```
Comment (justinswart) [inline Library/ViewModels/DashboardViewModel.swift]: Stray comment here.

File: Kickstarter-iOS/Views/Controllers/DashboardViewController.swift [hunk 4]
```diff
@@ -23,6 +25,10 @@ internal final class DashboardViewController: UITableViewController {
   internal override func viewDidLoad() {
     super.viewDidLoad()
 
+    self.tableView.backgroundView = self.backgroundView
+
+    self.tableView.insertSubview(self.loadingIndicatorView, aboveSubview: self.backgroundView)
```
Comment (stephencelis) [inline Kickstarter-iOS/Views/Controllers/DashboardViewController.swift]: Not sure if we do this elsewhere, but we gotta stop :(

From the iOS 11 release notes:

> Adding custom subviews directly to `UITableView` is not supported and might break in a future iOS release. (29094519)

We should start using container views to manage this.

File: Library/ViewModels/DashboardViewModel.swift (hunk 3 above)
Comment (stephencelis) [reply Library/ViewModels/DashboardViewModel.swift]: Beyond that, if we don't need it, we can drop the `Signal.merge` ;)

File: Kickstarter-iOS/Views/Controllers/DashboardViewController.swift (hunk 4 above)
Comment (stephencelis) [reply Kickstarter-iOS/Views/Controllers/DashboardViewController.swift]: (Alternatively we could use a table header view or cell to manage this.)

**Comment by justinswart** [comment]
@cdolm92 header's def a better approach 👍 let me know when you're done and I'll review again 🙂

**Comment by cdolm92** [comment]
@justinswart thanks! going to do some testing will let you know after 😄

File: Library/ViewModels/DashboardViewModelTests.swift [hunk 5]
```diff
@@ -10,38 +10,41 @@ import XCTest
 
 internal final class DashboardViewModelTests: TestCase {
   internal let vm: DashboardViewModelType = DashboardViewModel()
+
+  internal let animateOutProjectsDrawer = TestObserver<(), NoError>()
```
Comment (justinswart) [inline Library/ViewModels/DashboardViewModelTests.swift]: Thanks for alphabetizing 🙂

**Pull request merged**